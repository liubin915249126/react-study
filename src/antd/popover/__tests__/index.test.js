import React from 'react';
import { render, mount } from 'enzyme';
import Popover from '..';
import mountTest from '../../../tests/shared/mountTest';
import ConfigProvider from '../../config-provider';

describe('Popover', () => {
  mountTest(Popover);

  it('should show overlay when trigger is clicked', () => {
    const ref = React.createRef();

    const popover = mount(
      <Popover ref={ref} content="console.log('hello world')" title="code" trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    expect(ref.current.getPopupDomNode()).toBe(null);

    popover.find('span').simulate('click');
    expect(popover.find('Trigger PopupInner').props().visible).toBeTruthy();
  });

  it('shows content for render functions', () => {
    const renderTitle = () => 'some-title';
    const renderContent = () => 'some-content';
    const ref = React.createRef();

    const popover = mount(
      <Popover ref={ref} content={renderContent} title={renderTitle} trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    popover.find('span').simulate('click');

    const popup = ref.current.getPopupDomNode();
    expect(popup).not.toBe(null);
    expect(popup.innerHTML).toContain('some-title');
    expect(popup.innerHTML).toContain('some-content');
    expect(popup.innerHTML).toMatchSnapshot();
  });

  it('handles empty title/content props safely', () => {
    const ref = React.createRef();

    const popover = mount(
      <Popover trigger="click" ref={ref}>
        <span>show me your code</span>
      </Popover>,
    );

    popover.find('span').simulate('click');

    const popup = ref.current.getPopupDomNode();
    expect(popup).not.toBe(null);
    expect(popup.innerHTML).toMatchSnapshot();
  });

  it('props#overlay do not warn anymore', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

    const overlay = jest.fn();
    mount(
      <Popover content="console.log('hello world')" title="code" trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    expect(errorSpy.mock.calls.length).toBe(0);
    expect(overlay).not.toHaveBeenCalled();
  });

  it(`should be rendered correctly in RTL direction`, () => {
    const wrapper = mount(
      <ConfigProvider direction="rtl">
        <Popover title="RTL" visible>
          <span>show me your Rtl demo</span>
        </Popover>
      </ConfigProvider>,
    );
    expect(render(wrapper)).toMatchSnapshot();
  });
});
