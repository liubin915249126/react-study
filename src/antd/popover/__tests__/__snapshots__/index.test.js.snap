// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover handles empty title/content props safely 1`] = `"<div class=\\"ant-popover-content\\"><div class=\\"ant-popover-arrow\\"><span class=\\"ant-popover-arrow-content\\"></span></div><div class=\\"ant-popover-inner\\" role=\\"tooltip\\"><div class=\\"ant-popover-inner-content\\"></div></div></div>"`;

exports[`Popover should be rendered correctly in RTL direction 1`] = `
Array [
  <span
    class="ant-popover-open"
  >
    show me your Rtl demo
  </span>,
  <div>
    <div
      class="ant-popover ant-popover-rtl"
      style="opacity:0;pointer-events:none"
    >
      <div
        class="ant-popover-content"
      >
        <div
          class="ant-popover-arrow"
        >
          <span
            class="ant-popover-arrow-content"
          />
        </div>
        <div
          class="ant-popover-inner"
          role="tooltip"
        >
          <div
            class="ant-popover-title"
          >
            RTL
          </div>
          <div
            class="ant-popover-inner-content"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Popover shows content for render functions 1`] = `"<div class=\\"ant-popover-content\\"><div class=\\"ant-popover-arrow\\"><span class=\\"ant-popover-arrow-content\\"></span></div><div class=\\"ant-popover-inner\\" role=\\"tooltip\\"><div class=\\"ant-popover-title\\">some-title</div><div class=\\"ant-popover-inner-content\\">some-content</div></div></div>"`;
